### PR TITLE
Add tests for the Amqp project

### PR DESCRIPTION
**Pull request by jfonseca-aneo**
Resolves https://github.com/aneoconsulting/ArmoniK/issues/363

### CODE REVIEW

File: Adaptors/Amqp/src/QueueStorage.cs [hunk 1]
```diff
@@ -182,6 +183,12 @@ public ValueTask<bool> Check(HealthCheckTag tag)
                      priority / MaxInternalQueuePriority,
                      priority % MaxInternalQueuePriority);
 
+    if (priority > MaxPriority || priority > MaxInternalQueuePriority )
```
Comment (lemaitre-aneo) [inline Adaptors/Amqp/src/QueueStorage.cs:186]: `priority` can be larger than `MaxInternalQueuePriority`. This the sole purpose of multiple queues here.

File: Adaptors/Amqp/src/ServiceCollectionExt.cs [hunk 2]
```diff
@@ -105,9 +105,12 @@ public static class ServiceCollectionExt
       }
 
       var sessionProvider = new SessionProvider(amqpOptions,
-                                                logger);
+                                              logger);
```
Comment (lemaitre-aneo) [inline Adaptors/Amqp/src/ServiceCollectionExt.cs:108]: format?

File: Adaptors/Amqp/src/ServiceCollectionExt.cs [hunk 3]
```diff
@@ -105,9 +105,12 @@ public static class ServiceCollectionExt
...
 
-      serviceCollection.AddSingleton(sessionProvider);
+      serviceCollection.AddSingleton<IProviderBase<Session>, SessionProvider>(sp =>
+        {
```
Comment (lemaitre-aneo) [inline Adaptors/Amqp/src/ServiceCollectionExt.cs:111]: Couldn't you just write: `sp => sessionProvider` without having a block?

File: Adaptors/Amqp/tests/QueueStorageTests.cs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+  [Test]
+  public async Task SimpleBrokerTest()
+  {
+    var timeout = TimeSpan.FromMilliseconds(100);
+    await using var helper = new SimpleAmqpClientHelper();
+
+    var sender_ = new SenderLink(helper.Session, "sender-link", "q1");
+    var sendMsg = new Message("Hello AMQP!");
+    await sender_.SendAsync(sendMsg)
+      .ConfigureAwait(false);
```
Comment (lemaitre-aneo) [inline Adaptors/Amqp/tests/QueueStorageTests.cs:84]: [format] alignment

File: Adaptors/Amqp/tests/QueueStorageTests.cs [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  public void SetDefaultOptions()
+  {
+    /* These options are only to feed the QueueStorage constructor
+     * and they do not play any role in the how the connection is created,
+     * the later is defined in the  SimpleAmqpClientHelper class */
+    Options_ = CreateDefaultOptions();
+  }
+
+  [Test]
+  public async Task SimpleBrokerTest()
```
Comment (lemaitre-aneo) [inline Adaptors/Amqp/tests/QueueStorageTests.cs:76]: This test does not test the adapter but only the library. Is it on purpose?

File: Common/src/Injection/IProviderBase.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2021-$CURRENT_YEAR$. All rights reserved.
```
Comment (lemaitre-aneo) [inline Common/src/Injection/IProviderBase.cs:3]: `$CURRENT_YEAR$` not replaced

File: Common/src/Injection/ProviderBase.cs [hunk 7]
```diff
@@ -27,10 +27,10 @@
 
 namespace ArmoniK.Core.Common.Injection;
 
-public abstract class ProviderBase<T> : IHealthCheckProvider
+public abstract class ProviderBase<T> : IProviderBase<T>, IHealthCheckProvider
 {
   private readonly Func<Task<T>> builder_;
-  private          T?            object_;
+  private T? object_;
```
Comment (lemaitre-aneo) [inline Common/src/Injection/ProviderBase.cs:33]: I am pretty this does not follow the formatting rules.

File: Adaptors/Amqp/src/QueueStorage.cs (hunk 1 above)
Comment (jfonseca-aneo) [reply Adaptors/Amqp/src/QueueStorage.cs:186]: Without the exception we run into a index out bounds if a user inserts a `priority`  larger than `MaxInternalQueuePriority`, I think the idea is that in such case, we interpret that the message is reaaaally important for the user and put it in the last queue. I edited the code to reproduce that.

File: Adaptors/Amqp/src/ServiceCollectionExt.cs (hunk 3 above)
Comment (jfonseca-aneo) [reply Adaptors/Amqp/src/ServiceCollectionExt.cs:111]: Done

File: Adaptors/Amqp/tests/QueueStorageTests.cs (hunk 4 above)
Comment (jfonseca-aneo) [reply Adaptors/Amqp/tests/QueueStorageTests.cs:84]: This is already what the indent produces

File: Adaptors/Amqp/tests/QueueStorageTests.cs (hunk 5 above)
Comment (jfonseca-aneo) [reply Adaptors/Amqp/tests/QueueStorageTests.cs:76]: It was a play test, I'll remove it.

File: Common/src/Injection/IProviderBase.cs (hunk 6 above)
Comment (jfonseca-aneo) [reply Common/src/Injection/IProviderBase.cs:3]: Done

File: Common/src/Injection/ProviderBase.cs (hunk 7 above)
Comment (jfonseca-aneo) [reply Common/src/Injection/ProviderBase.cs:33]: This is already what the indent produces

**Review by lemaitre-aneo — DISMISSED** [review]

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**